lib/rules/unit-no-unknown/index.js

```diff
@@ -47,11 +43,14 @@ function rule(actual, options) {
 		}
 
 		function check(node, value, getIndex) {
+			if (node.type === 'decl' && !isStandardSyntaxDeclaration(node)) {
```

Comment:
Let's move this into the `walkDecls` at the bottom of the file, e.g.

```js
root.walkAtRules((atRule) => {
    if (!/^media$/i.test(atRule.name) && !atRule.variable) {
        return;
    }
    check(atRule, atRule.params, atRuleParamIndex);
});
root.walkDecls((decl) => {
    if (!isStandardSyntaxDeclaration(decl) {
        return;
    }
    check(decl, decl.value, declarationValueIndex));
}
```

So that it's grouped with the at-rule conditional.

---

Note: we should replace, in a separate refactor pull request, the `&& !atRule.variable` with the `isStandardSyntaxAtRule` util.

Reply:
Thanks for review :) I can change the `&& !atRule.variable` in separate pull request, do I need to create an issue for that?

Reply:
> I can change the `&& !atRule.variable` in separate pull request, do I need to create an issue for that?

Fantastic! No need to create a new issue. Just link to this thread in the pull request note.